doc/compiling/macos.md

```diff
@@ -34,12 +36,65 @@ cmake .. \
 make -j$(sysctl -n hw.logicalcpu)
 make install
 
-# M1 Macs w/ MacOS >= BigSur
-codesign --force --deep -s - macos/minetest.app
+# Apple Silicon (M1/M2) Macs w/ MacOS >= BigSur signature for local run
+codesign --force --deep -s - --entitlements ../misc/macos/entitlements/debug.entitlements macos/luanti.app
 ```
 
-## Run
+If you are using LuaJIT with `MAP_JIT` support use `debug_map_jit.entitlements`.
+
+### Run
 
 ```
-open ./build/macos/minetest.app
+open ./build/macos/luanti.app
 ```
+
+## Building for distribution
+
+### Generate Xcode project
+
+```bash
+mkdir build
+cd build
+
+cmake .. \
+    -DCMAKE_FIND_FRAMEWORK=LAST \
+    -DRUN_IN_PLACE=FALSE -DENABLE_GETTEXT=TRUE \
+    -DFREETYPE_LIBRARY=/path/to/lib/dir/libfreetype.a \
+    -DGETTEXT_INCLUDE_DIR=/path/to/include/dir \
+    -DGETTEXT_LIBRARY=/path/to/lib/dir/libintl.a \
+    -DLUA_LIBRARY=/path/to/lib/dir/libluajit-5.1.a \
+    -DOGG_LIBRARY=/path/to/lib/dir/libogg.a \
+    -DVORBIS_LIBRARY=/path/to/lib/dir/libvorbis.a \
+    -DVORBISFILE_LIBRARY=/path/to/lib/dir/libvorbisfile.a \
+    -DZSTD_LIBRARY=/path/to/lib/dir/libzstd.a \
+    -DGMP_LIBRARY=/path/to/lib/dir/libgmp.a \
+    -DJSON_LIBRARY=/path/to/lib/dir/libjsoncpp.a \
+    -DENABLE_LEVELDB=OFF \
+    -DENABLE_POSTGRESQL=OFF \
+    -DENABLE_REDIS=OFF \
+    -DJPEG_LIBRARY=/path/to/lib/dir/libjpeg.a \
+    -DPNG_LIBRARY=/path/to/lib/dir/libpng.a \
+    -DCMAKE_EXE_LINKER_FLAGS=-lbz2\
+    -GXcode
```

Comment:
I don't think we need to repeat all the `.a` stuff here.

Reply:
This is a minimal `.a` stuff configuration to use. If I remove part of it, everyone has to study which `.a` libs he has to use.